### PR TITLE
Prevent captcha field from setup in Joomla Admin

### PR DESCRIPTION
**Pull request by roland-d**
This is a redo of PR #7840 made by @ryandemmer 

It is possible, under certain circumstances (for example when JForm::getFieldset is called), for JFormFieldCaptcha to be initiated in the Joomla Administration. This causes a fatal error as  JFactory::getApplication()->getParams() (called on line 94), is not available in JApplicationAdministrator.

Line 88-89 added in this PR ends the setup process in this case.

I don't know of a use case for captcha in the Joomla Admin, but if there is one, then 

JFactory::getApplication()->getParams()->get('captcha', JFactory::getConfig()->get('captcha'));

needs to be altered accordingly.
### Testing

This is difficult to test as the error occurs in few circumstances, but was noticed in JCalPro following a recent JCE update which included a system plugin that uses JForm::getFieldset to get a list of all JFormField elements loaded. When editing a JCalPro Event, the form can include a captcha field in the front-end. When editing the Event in JCalPro in the Joomla Admin, the captcha field is not used, but  JForm::getFieldset will still initiate it, resulting in the error described above.

A test could be created using the Joomla Article Manager by adding the following field to /administrator/components/com_content/models/forms/article.xml:

``` xml
<fieldset
            name="captcha"
            label="Test Captcha"
        >
            <field
                name="captcha"
                type="captcha"
                label="Test Captcha"
                description="Test Captcha"
                validate="captcha"
            />
        </fieldset>
```

Then creating or editing an article with the latest version of JCE installed.

### CODE REVIEW

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on aefbee58e345263d3bba1e2ed8ccdf2affcbe208<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by gwsdesk** [comment]
When I add the code to the xml my ISIS template code is lost when I go to "Add article" in Joomla 3.5-RC4 (which is an update from 3.4.8 --> J3.5-Beta 1/2/3/4)-RC3/4

isis-missing

**Comment by roland-d** [comment]
@gwsdesk Not sure where you added the XML code in the template but in the instructions it is suggested to put it in the article.xml. However what you see seems to indicate the patch is not in place as this is what happens without the patch.

**Comment by gwsdesk** [comment]
Roland,
The code is perfectly well in place. See 
http://screencast.com/t/OJEz9rUmsDG as proof

File: libraries/cms/captcha/captcha.php [hunk 1]
```diff
@@ -79,6 +79,11 @@ class JCaptcha extends JObject
 	 */
 	public function __construct($captcha, $options)
 	{
+		if (JFactory::getApplication()->isAdmin())
```
Comment (wilsonge) [inline libraries/cms/captcha/captcha.php]: I think we can remove this now roland?

**Comment by wilsonge** [comment]
Leo can you please turn up error reporting to development - I'm sure there is another fatal error somewhere - and need to see where :) thanks!

**Comment by brianteeman** [comment]
@gwsdesk can you reply to @wilsonge question please<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by brianteeman** [comment]
I have tested this item :white_check_mark: successfully on e2b2586312948dab3689c5a0830fce4c836225ea<br /><br />Retested with current staging and current JCE release (2.5.16) confirmed fatal error
Applied patch and all good<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by justinherrin** [comment]
I have tested this item :red_circle: unsuccessfully on e2b2586312948dab3689c5a0830fce4c836225ea<br /><br />I am receiving a Fatal Error when trying to save a new article.

Fatal error: Call to undefined method JApplicationAdministrator::getParams() in /Applications/MAMP/htdocs/j3_githubtesting/libraries/cms/form/rule/captcha.php on line 38

Steps I took...
- Installed latest version of JCE (2.5.16)
- Assigned JCE to be my accounts editor
- Added your "Test Captcha" XML code to the article.xml file
- Tried to create a new article, received Fatal Error
- Applied your patch
- Was able to open the new article edit page
- Tried to save (or save & close) the new article, received Fatal Error listed above<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by roland-d** [comment]
@justinherrin That was a good test, I was able to check it and reproduce it. The latest change should solve the issue you found.

**Comment by justinherrin** [comment]
I have tested this item :white_check_mark: successfully on 324dde0acfa023fb15086030d4d4fbe7a63697cf<br /><br />Much better now :smile: <hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by laoneo** [comment]
I have tested this item :white_check_mark: successfully on 324dde0acfa023fb15086030d4d4fbe7a63697cf<br /><br />Tested it without JCE. Just put the XML

```xml
<fieldset name="captcha" label="Test Captcha" >
            <field
                name="captcha"
                type="captcha"
                label="Test Captcha"
                description="Test Captcha"
                validate="captcha"
            />
</fieldset>
```
inside the attribs fields, for example after line 610 in the article.xml file.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by brianteeman** [comment]
Rtc<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/8657">issues.joomla.org/joomla-cms/8657</a>.</sub>

**Comment by wilsonge** [comment]
Mergers. Please hold on merging this until I talk to Roland about this tomorrow. Thanks

**Comment by wilsonge** [comment]
Much better now! Thanks!

**Comment by Kubik-Rubik** [comment]
Thank you @roland-d!

**Pull request merged**